src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -121,24 +175,40 @@ public ParseStatus parseJavadocAsDetailNode(DetailAST javadocCommentAst) {
             result.setTree(tree);
         }
         catch (ParseCancellationException | IllegalArgumentException ignored) {
-            // If syntax error occurs then message is printed by error listener
+            // If syntax error occurs then error listeners are accordingly notified
             // and parser throws this runtime exception to stop parsing.
             // Just stop processing current Javadoc comment.
-            final ParseErrorMessage parseErrorMessage = errorListener.getErrorMessage();
-            result.setParseErrorMessage(parseErrorMessage);
         }
 
+        // Error listener is also notified of non-tight HTML in javadoc
+        // if errorListenerOption is true
+        result.setParseErrorMessage(errorListener.getErrorMessage());
+
         return result;
     }
 
     /**
      * Parses block comment content as javadoc comment.
      * @param blockComment
      *        block comment content.
+     * @param errorListenerOption Supplied by
+     *     {@link #parseJavadocAsDetailNode(DetailAST, ErrorListenerOption)} when it calls this
+     *     method. The javadoc grammar is designed in such a way to enable the JavadocParser to
+     *     store the first non-Tight HTML tag it encounters. If the javadoc being parsed indeed
+     *     has non-tight HTML tag(s) then the error listeners attached to ANTLR parser shall
+     *     accordingly be notified of the first non-tight HTML tag encountered provided
+     *     {@link ErrorListenerOption#NOTIFY_FOR_NON_TIGHT_HTML} is the argument supplied.
      * @return parse tree
-     * @noinspection deprecation
```

Comment:
Team city was giving errors about `redundant suppression`. Don't know why. Perhaps there was no such error from teamcity when this code was merged ? Do we control teamcity config ? Was it changed after this addition was made to the file ?